File: 04_BehaviouralDesignPatterns/02_ObserverPattern/cpp/badCode.cpp
#include <iostream>
#include <string>

class YouTubeChannel {
public:
    void uploadNewVideo(const std::string& videoTitle) {
        std::cout << "New video uploaded: " << videoTitle << std::endl;
        std::cout << "All subscribers will be notified of the new video upload." << std::endl;
        std::cout << "Sending notifications to all subscribers..." << std::endl;
        std::cout << "Email sent to subscribers." << std::endl;
    }
};

int main() {
    YouTubeChannel channel;
    channel.uploadNewVideo("Design Patterns in C++");
    channel.uploadNewVideo("Observer Pattern Explained");
    channel.uploadNewVideo("Iterator Pattern in Depth");
    
    return 0;
}

File: 04_BehaviouralDesignPatterns/02_ObserverPattern/badCode.java
class YouTubeChannel {
    public void uploadNewVideo(String videoTitle) {
        System.out.println("New video uploaded: " + videoTitle);
        System.out.println("All subscribers will be notified of the new video upload.");
        System.out.println("Sending notifications to all subscribers...");
        System.out.println("Email sent to subscribers.");
    }
}

public class badCode {
    public static void main(String[] args) {
        YouTubeChannel channel = new YouTubeChannel();
        channel.uploadNewVideo("Design Patterns in Java");
        channel.uploadNewVideo("Observer Pattern Explained");
        channel.uploadNewVideo("Iterator Pattern in Depth");
    }
}